apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/ServerTestUtils.java

```diff
@@ -399,6 +405,103 @@ public static List<Object> parseData(List<StaticRifResource> sampleResources) {
+   *
+   * @param addSamhsa if samhsa data should be added to the carrier test data
+   * @return the loaded records
+   */
+  public List<Object> loadSampleAData(boolean addSamhsa) {
+
+    List<Object> loadedRecords =
+        ServerTestUtils.get()
+            .loadData(Arrays.asList(StaticRifResourceGroup.SAMPLE_A.getResources()));
+    if (addSamhsa) {
+      addSamhsaToLoadedRecords(loadedRecords);
+    }
+
+    return loadedRecords;
+  }
+
+  /**
+   * Load the SAMPLE_A resources and then tweak carrier claim types to have a SAMHSA diagnosis code.
+   *
+   * @param loadedRecords the loaded records
+   */
+  private void addSamhsaToLoadedRecords(List<Object> loadedRecords) {
+    // Load the SAMPLE_A resources normally.
+    EntityManager entityManager = null;
+
+    try {
+      EntityManagerFactory entityManagerFactory =
+          PipelineTestUtils.get().getPipelineApplicationState().getEntityManagerFactory();
+      entityManager = entityManagerFactory.createEntityManager();
+
+      // Tweak the SAMPLE_A claims such that they are SAMHSA-related.
+      adjustCarrierClaimForSamhsaDiagnosis(loadedRecords, entityManager);
+
+    } finally {
+      if (entityManager != null && entityManager.getTransaction().isActive())
+        entityManager.getTransaction().rollback();
+      if (entityManager != null) entityManager.close();
+    }
+  }
+
+  /**
+   * Adjusts the carrier claim to support samhsa.
+   *
+   * @param loadedRecords the loaded records
+   * @param entityManager the entity manager
+   */
+  private void adjustCarrierClaimForSamhsaDiagnosis(
+      List<Object> loadedRecords, EntityManager entityManager) {
+
+    CarrierClaim carrierRifRecord =
+        loadedRecords.stream()
+            .filter(r -> r instanceof CarrierClaim)
+            .map(CarrierClaim.class::cast)
+            .findFirst()
+            .get();
+
+    entityManager.getTransaction().begin();
+    carrierRifRecord = entityManager.find(CarrierClaim.class, carrierRifRecord.getClaimId());
+    carrierRifRecord.setDiagnosis2Code(
+        Optional.of(Stu3EobSamhsaMatcherTest.SAMPLE_SAMHSA_ICD_9_DIAGNOSIS_CODE));
+    carrierRifRecord.setDiagnosis2CodeVersion(Optional.of('9'));
+    entityManager.merge(carrierRifRecord);
+    entityManager.getTransaction().commit();
+  }
+
+  /**
+   * Gets the pagination link by name from the response. Throws an exception if the given link name
+   * does not exist in the response.
+   *
+   * <p>Should be one of: next, previous, self, last, first
+   *
+   * @param response the response
+   * @param paginationLinkName the pagination link name
+   * @return the pagination link
+   */
+  public String getPaginationLink(Response response, String paginationLinkName) {
+
+    List<Map<String, ?>> links = response.jsonPath().getList("link");
+    return links.stream()
+        .filter(m -> m.get("relation").equals(paginationLinkName))
+        .findFirst()
+        .orElseThrow()
+        .get("url")
```

Comment:
I like the use of `orElseThrow()`to avoid calling two methods named `get` from two different classes here.